torch/linalg/__init__.py

```diff
@@ -88,6 +88,43 @@
 Alias of :func:`torch.det`.
 """)
 
+slogdet = _add_docstr(_linalg.linalg_slogdet, r"""
+linalg.slogdet(input) -> (Tensor, Tensor)
+
+Calculates the sign and natural logarithm of the absolute values of the determinants of the determinant(s)
+of a square matrix or batches of square matrices :attr:`input`.
+
+Supports input of ``float``, ``double``, ``cfloat`` and ``cdouble`` datatypes.
+
+.. note:: When given inputs on a CUDA device, this function synchronizes that device with the CPU.
+
+.. note:: For matrices that have zero determinant, this returns ``(0, -inf)``.
+          If :attr:`input` is batched then the entries in the result tensors corresponding to matrices with
+          the zero determinant have sign 0 and the natural logarithm of the absolute value of the determinant -inf.
+
+Arguments:
+    input (Tensor): the input matrix of size :math:`(n, n)` or the batch of matrices of size :math:`(*, n, n)`
+                    where `*` is one or more batch dimensions.
+
+Returns:
+    A namedtuple (sign, logabsdet) containing the sign of the determinant, and the natural logarithm
+    of the absolute value of determinant.
+
```

Comment:
One thing that the NumPy documentation briefly mentions but is absent here is the interpretation of sign. Maybe a note on how to interpret the sign value would be interesting?

Reply:
Is a sentence "The determinant can be computed with ``sign * exp(logabsdet)``." enough? I think having it we don't need to expand further what sign means and what values can it have for complex input or zero determinant matrices.

Reply:
OK, sounds good.